controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -362,21 +362,30 @@ func (r *NodeNetworkConfigurationPolicyReconciler) incrementUnavailableNodeCount
 
 func (r *NodeNetworkConfigurationPolicyReconciler) decrementUnavailableNodeCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy) {
 	policyKey := types.NamespacedName{Name: policy.GetName(), Namespace: policy.GetNamespace()}
+	err := tryDecrementingUnavailableNodeCount(r.Client, r.Client, policyKey)
+	if err != nil {
+		r.Log.Error(err, "error decrementing unavailableNodeCount with cached client, trying again with non-cached client.")
+		err = tryDecrementingUnavailableNodeCount(r.Client, r.APIClient, policyKey)
+		if err != nil {
+			r.Log.Error(err, "error decrementing unavailableNodeCount with non-cached client")
+		}
+	}
+}
+
+func tryDecrementingUnavailableNodeCount(cli client.Client, readerCli client.Reader, policyKey types.NamespacedName) error {
```

Comment:
Looks like controller runtime has the notion of client.Reader and client.Writer, can we replace "cli client.Client" with "writerCli client.Writer" ?

Reply:
Oddly, when I use `client.Writer` or `client.StatusWriter`, I'm getting an error saying that client.Writer/StatusWriter doesn't implement Status()

Reply:
Got it, `client.StatusWriter` is the correct type.

Reply:
I'm wondering if it's better to call the StatusWriter `cachedStatusWriter` to stress the point that one of the clients is cached and the other is not, but we don't have this notion used elsewhere so it may be just confusing...

Reply:
The fact that is cached is a caller detail not a callee.

Reply:
Updated